app/uk/gov/hmrc/vatregisteredcompanies/repositories/VatRegisteredCompaniesRepository.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.vatregisteredcompanies.repositories
+
+import java.time.Instant
+
+import javax.inject.{Inject, Singleton}
+import play.api.libs.json._
+import play.modules.reactivemongo.ReactiveMongoComponent
+import reactivemongo.api.commands.WriteResult
+import reactivemongo.api.indexes.{Index, IndexType}
+import reactivemongo.bson.BSONDateTime
+import reactivemongo.play.json.ImplicitBSONHandlers._
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.vatregisteredcompanies.models
+import uk.gov.hmrc.vatregisteredcompanies.models.{LookupResponse, Payload, VatNumber, VatRegisteredCompany}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import cats.implicits._
+
+final case class Wrapper(
+  vatNumber: VatNumber,
+  company: VatRegisteredCompany
```

Comment:
doesn't `VatRegisteredCompany` contain the `VatNumber` anyway?

Reply:
Yes, but I'm using the vat number as the index on the collection.